### ISSUE TITLE
Xampp error, apache keeps crashing 

### ISSUE THREAD

**Issue by pranjal-joshi97** [susanBuck/e2-fall20#93]
Hi all, 
I've been re-watching and re-tracing the steps to set up Zipfoods (just as a memory jog) but keep getting the following error in Xampp: 
![image](https://user-images.githubusercontent.com/42822597/99778122-d9198c00-2b45-11eb-806c-fe2da34a2960.png)

I've never gotten this before and I tried finding solutions online but they all involve changing up the port number on the local host and server. I could do this, but this would mean going back and changing the VirtualHost configs for p1, p2 and e2/practice files. 
Should I do this, or is there another solution? 
Has anyone experienced this before? Any ideas/advice on fixing this? 

Thanks in advance for the help!

**Comment by susanBuck** [comment]
Hi @pranjal-joshi97 - 

Before I'd mess with changing the port number, I'd try and identify/stop what might be using the port.

Details here on how to resolve a port conflict: 
https://hesweb.dev/e2/notes/local-server/xampp-windows#port-conflict

Let me know if you have any questions as you step through that process!

**Comment by pranjal-joshi97** [comment]
Hi @susanBuck the link was definitely helpful! Also for others facing the same issue, I found it easier to download CurrPorts using Firefox because for some reason Chrome was not downloading it properly. 

When running CurrPorts, I'm not seeing httpd.exe files and only the mysqld.exe files. No other files seem to be running on the 80 ports (at least none are popping up on CurrPorts).
![image](https://user-images.githubusercontent.com/42822597/99867789-5a792900-2bef-11eb-8272-3fb0be25ca05.png)

I'm running Xampp and CurrPorts as an Administrator.

**Comment by susanBuck** [comment]
Hi @pranjal-joshi97 - Thanks for going through those steps to cover our bases! 

One other thing to try is run `c:\xampp\apache\bin\httpd.exe -S` - this will reveal any issues that might exist in your Virtual Hosts file. You can share the output here so we can both double check it.

Some other "low hanging fruit" things to try is restarting your computer. This is such a broad/annoying troubleshooting suggestion, but sometimes it does the trick. : )

In the meantime, I'm booting up my Windows machine now to see if I can recreate the symptoms you're seeing.

**Comment by susanBuck** [comment]
Follow-up:

Can you scroll up in the xampp window and see if there's any additional output?

In my test, when I purposefully created a port conflict, it had more details above the same message you're seeing:

![image](https://user-images.githubusercontent.com/237711/99881321-09fdcd80-2be7-11eb-9492-e7265bc14795.png)

Finally, let's see if there are any additional hints in your Apache error log file. This file is located at `c:\xampp\apache\logs\error.log`. What I like to do is clear the file first of any previous errors, then attempt to start Apache - this just narrows down the feedback in case there was a lot of previous info recorded in the error log.

**Comment by pranjal-joshi97** [comment]
@susanBuck I ran `c:\xampp\apache\bin\httpd.exe -S` on my windows cmdr and got the following:
![image](https://user-images.githubusercontent.com/42822597/99892490-f2d2e480-2ca7-11eb-8f41-6bb82ee13152.png)
There seems to be a syntax error but I've never gotten this before when working on e2p1 and e2p2. Should I go in my Virtual Hosts and change up the syntax? 

I don't seem to be getting any port conflict error (like your screenshot) 
![image](https://user-images.githubusercontent.com/42822597/99892577-b653b880-2ca8-11eb-9a9e-9d81660ba4b7.png)

I cleared and looked through my error.log for apache and nothing came up as I re-ran Xampp
![image](https://user-images.githubusercontent.com/42822597/99892602-f3b84600-2ca8-11eb-9ec4-9f1dcaf1ba84.png)
But before clearing out the error.log I noticed that the last error entry happened on Nov. 3rd (when working on p2). Here are my access.logs just in case: 
![image](https://user-images.githubusercontent.com/42822597/99892618-25c9a800-2ca9-11eb-8479-ffd13c2eaf02.png)

**Comment by pranjal-joshi97** [comment]
I also looked through these online articles to figure out what's running on my 80 port:

https://stackoverflow.com/questions/1960750/how-do-i-find-which-program-is-using-port-80-in-windows#:~:text=6%20Answers&text=Start%20menu%20%E2%86%92%20Accessories%20%E2%86%92,and%20443%20for%20incoming%20connections. 

https://knowledge.autodesk.com/support/vault-products/troubleshooting/caas/CloudHelp/cloudhelp/2019/ENU/Vault-Install/files/GUID-1285D1B1-B74D-4DD0-8C8E-F1F64F8B744D-htm.html

I used the command `netstat -aon | findstr :80` and got the following results: 
![image](https://user-images.githubusercontent.com/42822597/99892664-be602800-2ca9-11eb-912d-45d4a29b98ba.png)

I don't know how accurate or helpful this is for this situation but just wanted to share other resources I've tried out.

**Comment by susanBuck** [comment]
Hi @pranjal-joshi97 -

Sounds like there's something wrong with the configs based on the output you got from `c:\xampp\apache\bin\httpd.exe -S`:

![image](https://user-images.githubusercontent.com/237711/99992488-677b5100-2d84-11eb-88b0-bf2fe212c66c.png)


The error feedback is pointing to line 521 in httpd.conf (Apache config file), which according to my httpd.conf, is just the line that includes your VirtualHosts config file:

```
520: # Virtual hosts
521: Include conf/extra/httpd-vhosts.conf
```

I suspect the problem is in the VirtualHosts config file, and it's just getting flagged in the httpd conf file when it attempts to load the VirtualHosts config file.

Can you share the full contents of your VirtualHosts config file here (`c:\xampp\apache\conf\extra\httpd-vhosts.conf`)?

**Comment by pranjal-joshi97** [comment]
@susanBuck  I found out where the error was! I had accidently put an extra colon in my VirtualHosts config file. Here it is: 
![image](https://user-images.githubusercontent.com/42822597/100050431-fd2de380-2e4b-11eb-94ca-ef019804a133.png)

I fixed it and Apache is running smoothly now! I can't believe such a small mistake caused such a big problem haha. I'll have to be extra extra careful when adding in any projects to the VirtualHosts config file.  

Thank you so much for helping out :)
